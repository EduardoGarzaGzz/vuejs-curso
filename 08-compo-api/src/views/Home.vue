<template>
  <div class="home">
    <h1>Reactive vs Ref</h1>
		<h3>{{ ironman }}</h3>
		<h3>{{ hulk }}</h3>
		
		<button @click="changeIronman">changeIronman</button>
		<button @click="changeHulk">changeHulk</button>
		
		<h3>Ironman: {{ ironmanIsAlive }}</h3>
		<h3>Hulk: {{ hulkIsAlive }}</h3>
		
		<button @click="changeStatus">changeStatus</button>
  </div>
</template>

<script>

import { reactive, ref } from "vue";

export default {
	name: 'Home',
	setup() {
		const ironman = ref( { name: 'Tony', age: 30 } );
		const hulk    = reactive( { name: 'Bruce', age: 35 } );
		
		const ironmanIsAlive = reactive( false );
		const hulkIsAlive    = ref( true );
		
		const changeStatus = () => {
			ironmanIsAlive.value = !ironmanIsAlive.value;
			hulkIsAlive.value    = !hulkIsAlive.value;
		};
		
		return {
			ironman,
			hulk,
			ironmanIsAlive,
			hulkIsAlive,
			changeStatus,
			changeIronman() {
				ironman.value.name = 'Tony Stark';
			},
			changeHulk() {
				hulk.name = 'Bruce Banner';
				hulk.age  = 40;
			}
		}
	}
}
</script>
