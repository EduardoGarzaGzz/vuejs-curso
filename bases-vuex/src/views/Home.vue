<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">
    <Counter />
  </div>
</template>

<script lang="ts">
import { defineAsyncComponent } from 'vue';
import { Options, Vue }         from 'vue-class-component';

@Options( {
  components: {
    Counter: defineAsyncComponent( () => import('@/components/Counter.vue') )
  }
} )
export default class Home extends Vue {}
</script>
