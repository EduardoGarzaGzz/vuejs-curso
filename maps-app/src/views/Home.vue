<template>
  <div class="home">
    <MapView />
  </div>
</template>

<script lang="ts">

import MapViewComponent    from '@/components/map-view/map-view.component.vue';
import { usePlacesStore }  from '@/composables/usePlacesStore';
import { defineComponent } from 'vue';

export default defineComponent( {
	name      : 'HomeView',
	components: {
		MapView: MapViewComponent
	},
	setup() {
		usePlacesStore();
	}
} );
</script>
