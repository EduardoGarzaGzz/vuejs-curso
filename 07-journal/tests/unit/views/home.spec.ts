import Home             from '@/views/Home.vue';
import { shallowMount } from '@vue/test-utils';

describe( 'Test View: Home.vue', function() {
	it( 'debe de renderizar el componente correctamente', () => {
		const wrapper = shallowMount( Home );
		expect( wrapper.html() ).toMatchSnapshot();
	} );

	it( 'hacer click en un button debe de redireccionar a no-entry', () => {
		const $router = {
			push: jest.fn()
		};

		//@ts-ignore
		const wrapper = shallowMount( Home, {
			global: {
				mocks: {
					$router
				}
			}
		} );

		wrapper.find( 'button' ).trigger( 'click' );
		expect( $router.push ).toHaveBeenCalledTimes( 1 );
		expect( $router.push ).toHaveBeenCalledWith( { 'name': 'no-entry' } );
	} );
} );
