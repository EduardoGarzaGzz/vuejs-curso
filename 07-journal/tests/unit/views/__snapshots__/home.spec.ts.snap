// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test View: Home.vue debe de renderizar el componente correctamente 1`] = `<div class="home"><img alt="Vue logo"><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Success</button></div>`;
